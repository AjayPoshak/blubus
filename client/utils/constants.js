export const HOME_PG_API_URL = 'http://localhost:3004/search';
export const LISTING_API_URL = 'http://localhost:3004/listing';
export const SEATLAYOUT_API_URL = 'http://localhost:3004/seatlayout';

export const MAX_ITEMS_IN_SEARCH_RESULTS = 5;

// PATTERNS
export const mobilePattern = /^\d{10}$/;
export const numberPattern = /^\d{1,2}$/;
export const emailPattern = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
