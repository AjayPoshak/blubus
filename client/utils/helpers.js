/* global Raven, __ENV__ */
export default function logErrorToService(error, info) {
	if (typeof Raven !== 'undefined' && __ENV__ === 'PRODUCTION') {
		Raven.captureException(error, { extra: info });
	}
}

/**
 * Sets debouncing for function passed in callback
 * @param {*} callBack Function to be called after delay
 * @param {*} delay Time in milliseconds after which the function should be called
 */
export function debounce(callBack, delay) {
	let tId;
	return value => {
		clearTimeout(tId);
		tId = setTimeout(() => {
			callBack(value);
		}, delay || 300);
	};
}
