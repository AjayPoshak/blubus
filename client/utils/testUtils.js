import React from 'react';
import { MemoryRouter } from 'react-router';
import { mount, render, shallow } from 'enzyme';

export function fullRender(Component, props = {}) {
	return mount(
		<MemoryRouter>
			<Component {...props} />
		</MemoryRouter>
	);
}

export function staticRender(Component, props = {}) {
	return render(<Component {...props} />);
}

export function shallowRender(Component, props = {}) {
	return shallow(<Component {...props} />);
}
