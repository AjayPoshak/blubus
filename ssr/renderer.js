export const createScriptTag = script => `<script src=${script}></script>`;

const renderFullPage = (html, preloadedState, styles, bundles) => `<!DOCTYPE html>
	<html lang="en">
	
	<head>
		<meta charset="UTF-8">
		<title>Bus Booking in Africa</title>
		<meta http-equiv="X-UA-Compatible" content="ie=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<!-- multiple icons for IE -->
		<meta name="theme-color" content="#2396b0">
		<meta name="msapplication-square310x310logo" content="icon_largetile.png">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
		<meta name="description" content="online bus booking service in Rwanda, 
		book bus online from anywhere">
		<!-- icon in the highest resolution we need it for -->
		<!-- reuse same icon for Safari -->
		<link rel="manifest" href="../public/manifest.json">
		<link rel="apple-touch-icon" sizes="57x57" href="/apple-icon-57x57.png">
		<link rel="apple-touch-icon" sizes="60x60" href="/apple-icon-60x60.png">
		<link rel="apple-touch-icon" sizes="72x72" href="/apple-icon-72x72.png">
		<link rel="apple-touch-icon" sizes="76x76" href="/apple-icon-76x76.png">
		<link rel="icon" sizes="192x192" href="/public/images/BluBus_192x192.png">
		<link rel="icon" type="image/png" sizes="96x96" href="/favicon-96x96.png">
		<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
		<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
		<link rel="apple-touch-icon" sizes="114x114" href="/apple-icon-114x114.png">
		<link rel="apple-touch-icon" sizes="120x120" href="/apple-icon-120x120.png">
		<link rel="apple-touch-icon" sizes="144x144" href="/apple-icon-144x144.png">
		<link rel="apple-touch-icon" sizes="152x152" href="/apple-icon-152x152.png">
		<link rel="apple-touch-icon" sizes="180x180" href="/apple-icon-180x180.png">
		<link rel="apple-touch-startup-image" href="/public/images/BluBus_192x192.png">
		<link rel="icon" type="image/png" sizes="192x192"  href="/android-icon-192x192.png">
		<meta name="msapplication-TileColor" content="#ffffff">
		<meta name="msapplication-TileImage" content="/ms-icon-144x144.png">
		${styles}
	</head>
	
	<body>
		<div id="root">${html}</div>
	</body>
	<link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
	<script src=https://cdn.ravenjs.com/3.24.0/raven.min.js
 crossorigin=anonymous></script>
	<script>
		window.__PRELOADED_STATE__ = ${JSON.stringify(preloadedState).replace(/</g, '\\u003c')}
	</script>
	<script src=${bundles.manifest.js}></script>
	<script src=${bundles.main.js}></script>
	<script src=${bundles.vendors.js}></script>
	<script>
		if('serviceWorker' in navigator) {
			window.addEventListener('load', () => {
				navigator.serviceWorker.register('./service-worker.js')
			})
		}
	</script>
	<!-- Third Party CSS for calendar module -->
	<link rel="stylesheet" href="/public/rc-calendar.css">
	</html>`;
export default renderFullPage;
