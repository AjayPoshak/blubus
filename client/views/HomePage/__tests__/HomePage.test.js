import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { HomePage } from '../index';
import { fullRender } from '../../../utils/testUtils';

describe('Home Page component', () => {
	const mockData = {
		homePage: {
			searchResult: [{ city: 'Ajmer' }, { city: 'Mumbai' }, { city: 'Jaipur' }, { city: 'Amer' }]
		},
		dispatch: jest.fn()
	};
	let wrapper;

	beforeAll(() => {
		wrapper = fullRender(HomePage, mockData);
	});

	it('renders Header component', () => {
		expect(wrapper.find('Header').length).toBe(1);
	});

	it('has three input boxes', () => {
		expect(wrapper.find('input').length).toBe(3);
	});

	it('search button is rendered', () => {
		expect(wrapper.find('button').length).toBe(1);
	});

	it('first inputbox takes location input', () => {
		wrapper
			.find('input')
			.first()
			.simulate('change', {
				target: { value: 'a' }
			});
		const firstInput = wrapper.find('input').first();
		expect(firstInput.props().value).toBe('a');
	});

	it('search results are shown for location input', () => {
		wrapper
			.find('input')
			.first()
			.simulate('change', {
				target: { value: 'a' }
			});
		const firstInput = wrapper.find('input').first();
		expect(wrapper.find('SearchResults').length).toBe(1);
	});

	it('second inputbox takes destination input', () => {
		wrapper
			.find('input')
			.at(1)
			.simulate('change', {
				target: { value: 'm' }
			});
		expect(
			wrapper
				.find('input')
				.at(1)
				.props().value
		).toBe('m');
	});

	it('search results are shown for destination input', () => {
		/**
		 *  For first input click on SearchResults and simulate click
		 *  on first element so that SearchResults closes for first element
		 **/
		wrapper
			.find('SearchResults li')
			.first()
			.simulate('click');
		wrapper
			.find('input')
			.at(1)
			.simulate('change', {
				target: { value: 'm' }
			});
		expect(wrapper.find('SearchResults').length).toBe(1);
	});

	it('focus third inputbox opens calendar widget', () => {
		wrapper
			.find('input')
			.last()
			.simulate('focus');
		expect(wrapper.find('Calendar').length).toBe(1);
	});

	it("select today's date as journey date from calendar", () => {
		wrapper
			.find('input')
			.last()
			.simulate('focus');
		const today = new Date();
		const currentDate = ('0' + today.getDate()).slice(-2),
			currentMonth = ('0' + (today.getMonth() + 1)).slice(-2),
			currentYear = today.getFullYear(),
			currentJourneyDate = `${currentDate}/${currentMonth}/${currentYear}`;

		expect(wrapper.find('Calendar .rc-calendar-selected-day').text()).toBe(currentDate.toString());
		wrapper.find('Calendar .rc-calendar-selected-day').simulate('click');
		expect(
			wrapper
				.find('input')
				.last()
				.props().value
		).toBe(currentJourneyDate);
	});

	it('calendar closes after date input has been selected', () => {
		wrapper
			.find('input')
			.last()
			.simulate('focus');
		wrapper.find('Calendar .rc-calendar-selected-day').simulate('click');
		expect(wrapper.find('Calendar').length).toBe(0);
	});

	it('UI snapshot of homePage component', () => {
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
