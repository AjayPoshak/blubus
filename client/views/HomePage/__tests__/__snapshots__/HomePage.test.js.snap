// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page component UI snapshot of homePage component 1`] = `
<section>
  <section>
    <section
      className="style__HeaderWrapper-s1kbzlmn-0 kIONTA"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <header>
          <span
            className="logo__TextWrapper-nm9i2-0 huAJcX"
          >
            <img
              alt="blubus logo"
              className="logo__LogoImg-nm9i2-1 eMKueu"
              src="../../public/images/BluBus_192x192.png"
            />
            <h1
              className="logo__LogoText-nm9i2-2 bObqUd"
            >
              BluBus
            </h1>
          </span>
        </header>
      </a>
    </section>
    <article
      className="style__PageWrapper-xm59nd-0 dhcnhC"
    >
      <span
        className="style__InputWrapper-t4p2yc-1 eCZAey"
      >
        <input
          className="style__Input-t4p2yc-0 jGewFw"
          icon="https://res.cloudinary.com/ddbxa4afa/image/upload/v1527944418/blubus/location.svg"
          onChange={[Function]}
          placeholder="Choose your location"
          type="text"
          value=""
        />
        <img
          alt="https://res.cloudinary.com/ddbxa4afa/image/upload/v1527944418/blubus/location.svg"
          className="style__Icon-t4p2yc-2 gyKMQZ"
          src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1527944418/blubus/location.svg"
        />
      </span>
      <div
        style={
          Object {
            "marginBottom": "70px",
            "marginTop": "70px",
          }
        }
      >
        <span
          className="style__InputWrapper-t4p2yc-1 eCZAey"
        >
          <input
            className="style__Input-t4p2yc-0 jGewFw"
            icon="https://res.cloudinary.com/ddbxa4afa/image/upload/v1527944416/blubus/destination.svg"
            onChange={[Function]}
            placeholder="Choose your destination"
            type="text"
            value=""
          />
          <img
            alt="https://res.cloudinary.com/ddbxa4afa/image/upload/v1527944416/blubus/destination.svg"
            className="style__Icon-t4p2yc-2 gyKMQZ"
            src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1527944416/blubus/destination.svg"
          />
        </span>
      </div>
      <span
        className="style__InputWrapper-t4p2yc-1 eCZAey"
      >
        <input
          className="style__Input-t4p2yc-0 jGewFw"
          icon="https://res.cloudinary.com/ddbxa4afa/image/upload/v1531114873/blubus/calendar.svg"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Choose date for journey"
          type="text"
          value=""
        />
        <img
          alt="https://res.cloudinary.com/ddbxa4afa/image/upload/v1531114873/blubus/calendar.svg"
          className="style__Icon-t4p2yc-2 gyKMQZ"
          src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1531114873/blubus/calendar.svg"
        />
      </span>
      <div
        style={
          Object {
            "marginTop": "70px",
          }
        }
      >
        <button
          className="Button__ButtonStyle-gjwxrp-0 fiTpLU"
          onClick={[Function]}
        >
          Search Buses
        </button>
      </div>
    </article>
  </section>
</section>
`;
